TCAT/Controllers/HomeOptionsCardViewController.swift

```diff
@@ -0,0 +1,561 @@
+    let maxHeaderCount: CGFloat = 2
+    let maxSeparatorCount: CGFloat = 1
+    let maxScreenCoverage: CGFloat = 3/5
+    let maxRecentsCount = 2
+    let maxFavoritesCount = 2
+    
+    /** Returns the height of a card that would contain two favorites and two recent searches.
+     This is the height that we will cap all optionCards at, regardless of the phone. */
+    var maxCardHeight: CGFloat {
+        return tableViewRowHeight * maxRowCount + headerHeight * maxHeaderCount + searchBarHeight + HeaderView.separatorViewHeight * maxSeparatorCount
+    }
+    
+    /** Checks to see if the bottom of the card at maximum height would cover more than 3/5 of the screen */
+    var isDynamicSearchBar: Bool {
+        
+        return maxCardHeight > (UIScreen.main.bounds.height*maxScreenCoverage - HomeMapViewController.optionsCardInset.top)
+    }
+    var recentLocations: [Place] = [] {
+        didSet {
+            if recentLocations.count > maxRecentsCount {
+                recentLocations = Array(recentLocations.prefix(2))
+            }
+            if !isNetworkDown {
+                sections = createSections()
+            }
+        }
+    }
+    var favorites: [Place] = [] {
+        didSet {
+            if favorites.count > maxFavoritesCount {
+                favorites = Array(favorites.prefix(2))
+            }
+            if !isNetworkDown {
+                sections = createSections()
+            }
+        }
+    }
+    var sections: [Section] = [] {
+        didSet {
+            tableView.reloadData()
+            delegate?.updateSize()
+        }
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        setupTableView()
+        setupInfoButton()
+        setupSearchBarSeparator()
+        setupSearchBar()
+        
+        setupConstraints()
+        
+        updatePlaces()
+    }
+
+    func setupTableView() {
+        tableView = UITableView(frame: .zero, style: .grouped)
+        tableView.backgroundColor = view.backgroundColor
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.separatorStyle = .none
+        tableView.keyboardDismissMode = .onDrag
+        tableView.tableFooterView = UIView(frame: .zero)
+        tableView.showsVerticalScrollIndicator = false
+        tableView.rowHeight = tableViewRowHeight
+        tableView.register(PlaceTableViewCell.self, forCellReuseIdentifier: Constants.Cells.placeIdentifier)
+        tableView.register(GeneralTableViewCell.self, forCellReuseIdentifier: Constants.Cells.seeAllStopsIdentifier)
+        view.addSubview(tableView)
+    }
+    
+    func setupSearchBar() {
+        searchBar = UISearchBar()
+        searchBar.placeholder = Constants.General.searchPlaceholder
+        searchBar.delegate = self
+        searchBar.searchBarStyle = .default
+        let textFieldInsideSearchBar = searchBar.value(forKey: "searchField") as? UITextField
+        textFieldInsideSearchBar?.backgroundColor = .white
```

Comment:
Should use `Colors.white` from Style.swift, in case we ever choose to use the slightest off white